cpp/src/arrow/compute/kernels/vector_cumulative_ops_test.cc

```diff
@@ -0,0 +1,257 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include "arrow/array.h"
+#include "arrow/chunked_array.h"
+#include "arrow/testing/gtest_util.h"
+#include "arrow/testing/util.h"
+#include "arrow/type.h"
+
+#include "arrow/compute/api.h"
+#include "arrow/compute/kernels/test_util.h"
+
+namespace arrow {
+namespace compute {
+
+template <typename T, typename OptionsType>
+class TestCumulativeOp : public ::testing::Test {
```

Comment:
We've been inconsistent in this, but to minimize compile times it would be better to use [value parameterization](https://github.com/google/googletest/blob/main/docs/advanced.md#value-parameterized-tests) (or even manual parameterization using for loops) rather than type parameterization.

For example the "values" for value parameterization could be instances of a struct:
```c++
struct CumulativeSumParam {
  std::shared_ptr<DataType> type;
  std::string json_input;
  std::string json_start;
  std::string json_expected;
}
```